processor/main.py

```diff
@@ -51,7 +51,7 @@ def import_stuff():
 
 def extract_id_from_url(url):
     split = url.split("/")
-    id = split[-1].split('&')[0]
+    id = split[-1].split('&')[0].split('?si=')[0].split('?t=')[0].split('?feature=shared')[0]
```

Comment:
you can use the [urllib](https://docs.python.org/3/library/urllib.parse.html) native for this
something like
```py
urlData = urllib.parse.urlparse(url)
id = urllib.parse.parse_qs(urlData.query).get("v") # this can be null (if bad data), for youtu.be links you have to check the path
```